app/models/nic/interface.rb

```diff
@@ -44,6 +44,22 @@ def vlanid
       return ''
     end
 
+    def mtu
+      # Determine a mtu according to the following cascading rules:
+      # 1. if the interface has a v4 subnet with a non-blank mtu, use that
```

Comment:
As MTU is now a required attribute, this condition will never be false. So basically, this always prefers the IPv4 MTU. How about always taking the maximum or minimum value?

```ruby
[self.subnet.mtu, self.subnet6.mtu].select(&:present?).compact.max
```

The longer I think about that, I'm wondering if this is the right approach. The MTU is basically a setting on the ethernet layer. It defines the maximum size of the Ethernet frames. The subnets are on a layer on top the ethernet layer in the OSI model. Subnets are an IP thing.
I have a feeling, what we really need is to put this field on the nic model. We can define a default MTU for new hosts on the subnet. Better would be a dedicated "network" model that represents the ethernet layer or VLAN.

@ares, @sean797: What do you guys think?

Reply:
In general, thumbs up. But that would be quite a change to vlan interfaces if we define vlan through network. Also would it be required for every user to define a network, or only in case they want to customize mtu/vlan on one place for all interfaces?